File: java/src/main/java/mx/iteso/ut/QuesoManchego.java
package mx.iteso.ut;

/**
 * clase manchego.
 */
public class QuesoManchego implements Queso {

    /**
     *
     * @return esta derretido?.
     */
    public boolean isMelted() {
        return false;
    }

    /**
     *
     * @return temperatura.
     */
    public int getCurrentTemperature() {
        return 0;
    }

    /**
     *
     * @return temperatura.
     */
    public int getMeltingTemperature() {
        return 0;
    }

    /**
     *
     * @param temp temperatura de la quesadilla.
     */
    public void setCurrentTemperature(final int temp) {

    }

    /**
     *
     * @param melted si el queso esta derretido.
     */
    public void melt(final boolean melted) {

    }
}

File: csharp/unittest-practice/Clases/QuesoManchego.cs
namespace unittestpractice.Clases
{
    public class QuesoManchego :  IQueso
    {
        private bool _melted;
        private int _temperature;
        private const int Melting = 25;

        public bool isMelted()
        {
            return _melted;
        }

        public int getCurrentTemperature()
        {
            return _temperature;
        }

        public int getMeltingTemperature()
        {
            return Melting;
        }

        public void setCurrentTemperature(int pTemp)
        {
            _temperature = pTemp;
        }

        public void melt(bool pMelted)
        {
            _melted = pMelted;

        }
    }
}
